lib/shopify_api/rest.ex

```diff
@@ -11,25 +11,76 @@ defmodule ShopifyAPI.REST do
   alias ShopifyAPI.JSONSerializer
   alias ShopifyAPI.REST.Request
 
-  @doc false
-  def get(%AuthToken{} = auth, path, params \\ []) do
-    Request.perform(auth, :get, path, "", params)
+  @doc """
+  Underlying utility retrieval function. The options passed affect both the
+  return value and, ultimately, the number of requests made to Shopify.
+
+  ## Options
+
+    `:pagination` - Can be `:none`, `:stream`, or `:auto`. Defaults to :auto
+    `:none` will only return the first page. You won't have access to the headers to manually
+      paginate.
+    `:auto` will block until all the pages have been retrieved and concatenated together.
+    `:stream` will return a `Stream`, prepopulated with the first page.
```

Comment:
Thanks for adding these, they're useful. I appreciate the note that we still don't get headers from `:none`.

🛠 Maybe it's best to list `:auto` first, since it's the default?